lib/generate_rota.rb

```diff
@@ -52,7 +53,7 @@ def fill_slots(people, slots_to_fill, roles_config = {})
         end
 
         # prefer assigning shift to devs who have been given fewer shifts so far, or less burdensome shifts
-        assignable_devs = remaining_devs.sort_by { |person| fairness_calculator.weight_of_shifts(person.assigned_shifts) }
+        assignable_devs = remaining_devs.sort_by { |person| [fairness_calculator.weight_of_shifts(person.assigned_shifts), person.name] }
```

Comment:
Looks like this slipped in by accident before being changed in the final commit? Too late to rebase now obviously, but would be good to confirm whether this change here was intentional.

Reply:
No - it was a change I added in to test that the undefined sort order was the problem, but after I knew that randomised secondary order would solve the problem better, I intended to remove. I should have removed in an intermediate commit I could squash together with commit 2.